docs/csharp/programming-guide/concepts/expression-trees/how-to-use-expression-trees-to-build-dynamic-queries.md

```diff
@@ -1,107 +1,266 @@
+
+```csharp
+var companyNames = new[] {
+    "Consolidated Messenger", "Alpine Ski House", "Southridge Video",
+    "City Power & Light", "Coho Winery", "Wide World Importers",
+    "Graphic Design Institute", "Adventure Works", "Humongous Insurance",
+    "Woodgrove Bank", "Margie's Travel", "Northwind Traders",
+    "Blue Yonder Airlines", "Trey Research", "The Phone Company",
+    "Wingtip Toys", "Lucerne Publishing", "Fourth Coffee"
+};
+
+// We're using an in-memory array as the data source, but the IQueryable could have come
+// from anywhere -- an ORM backed by a database, a web request, or any other LINQ provider.
+IQueryable<string> companyNamesSource = companyNames.AsQueryable();
+
+var qry = companyNames.OrderBy(x => x);
+```
+
+Every time you run this code, the same exact query will be executed. This is frequently not very useful, as you may want your code to execute different queries under varying conditions. This topic describes how you can execute a different query based on runtime state.
+
+## IQueryable, IQueryable\<T> and expression trees
+
+Fundamentally, an <xref:System.Linq.IQueryable> has two components:
+
+* <xref:System.Linq.IQueryable.Expression>&mdash;a language- and datasource-agnostic representation of the current query's elements, in the form of an expression tree.
+* <xref:System.Linq.IQueryable.Provider>&mdash;an object which knows how to translate the current query into actual .NET objects, a.k.a. an instance of a LINQ provider.
+
+In the context of dynamic querying, the provider will usually remain the same; the expression tree of the query will be different from query to query.
+
+Because expression trees are immutable, if you want a different expression tree&mdash;and thus a different query&mdash;you'll need to translate the existing expression tree to a new one, and thus to a new **IQueryable**.
+
+The following sections describe specific techniques for querying differently in response to runtime state:
+
+- Referencing runtime state directly in the expression tree
+- Calling additional LINQ methods
+- Varying the expression tree passed into the LINQ methods
+- Constructing expression trees using the factory methods at <xref:System.Linq.Expressions.Expression>
+
+## Referencing runtime state from the expression tree
+
+Assuming the LINQ provider supports it, the simplest way to query dynamically is to reference the runtime state directly in the query, via a closed-over variable:
+
+```csharp
+var length = 1;
+var qry = companyNamesSource
+    .Select(x => x.Substring(0, length))
+    .Distinct();
+    
+Console.WriteLine(string.Join(",", qry));
+// prints: C, A, S, W, G, H, M, N, B, T, L, F
+
+length = 2;
+Console.WriteLine(string.Join(",", qry));
+// prints: Co, Al, So, Ci, Wi, Gr, Ad, Hu, Wo, Ma, No, Bl, Tr, Th, Lu, Fo 
+```
+
+Note that the expression tree hasn't been modified; the query returns different values only because the value of `length` has been changed.
```

Comment:
nit:

```suggestion
The expression tree hasn't been modified; the query returns different values only because the value of `length` has been changed.
```